In [1]:
import pandas as pd

# Importing data from each year and creating one large data file

#### Looking at what is in each file using 2012 as a baseline in order to decide what to merge and how

In [2]:
Incident_2012 = pd.read_csv('../theft_and_crime_clt/Data_input/2012/2012_Incident.csv')
Incident_2012.head()

Complaint_No Block_No Direction  \
0  20120616144301     9108             
1  20120616144400     2903        N    
2  20120616144700      601        W    
3  20120616145104     5522             
4  20120616145201     7718             

                                Street_Name Street_Type    Suffix      Apt_No  \
0  REAMES                                            RD                         
1  DAVIDSON                                          ST                         
2  24TH                                              ST                         
3  LABORDE                                           AV                         
4  PRINDLE LAKE                                      DR                         

                                       City State Zipcode  ...  \
0  CHARLOTTE                                   NC   28216  ...   
1  CHARLOTTE                                   NC   28205  ...   
2  CHARLOTTE                                   NC   28206  ...   
3  CHARLOTTE                                   NC   28269  ...   
4  CHARLOTTE                                   NC   28227  ...   

  Incident_to_date Incident_to_Time                          Place1  \
0          6/16/12           1330.0  Residential                      
1          6/16/12           1230.0  Residential                      
2              NaN              NaN  Commercial Place                 
3          6/15/12           1635.0  Residential                      
4          6/16/12           1441.0  Residential                      

                                              Place2  \
0  Residential Care Facility                     ...   
1  Apartment/Duplex Private Res                  ...   
2  Other - Commercial Place                      ...   
3  Private Residence                             ...   
4  Private Residence                             ...   

                                    Clearance_Status Clearance_Date  \
0  Exceptionally Cleared - Located (Missing Perso...        6/16/12   
1  Open                                          ...                  
2  Open                                          ...                  
3  Open                                          ...                  
4  Exceptionally Cleared - Victim Chose not to Pr...        6/18/12   

                 Case_Status                               Reporting_Agency  \
0  Close/Cleared              Charlotte-Mecklenburg Police Department         
1  Further Investigation      Charlotte-Mecklenburg Police Department         
2  Inactive                   Charlotte-Mecklenburg Police Department         
3  Inactive                   Charlotte-Mecklenburg Police Department         
4  Close/Cleared              Charlotte-Mecklenburg Police Department         

                                   Follow_up_Section  \
0  70-FIB - MISSING PERSONS                      ...   
1  06-EASTWAY DIVISION                           ...   
2  02-METRO DIVISION                             ...   
3  11-NORTH DIVISION                             ...   
4  12-HICKORY GROVE DIVISION                     ...   

                                      NIBRS_Hi_Class  
0  Missing Person                                ...  
1  Theft of Motor Vehicle Parts from Vehicle     ...  
2  All Other Thefts                              ...  
3  Damage/Vandalism Of Property                  ...  
4  Intimidation                                  ...  

[5 rows x 28 columns]

In [3]:
Offenses_2012 = pd.read_csv('../theft_and_crime_clt/Data_input/2012/2012_Offenses.csv')
Offenses_2012.head()

Complaint_No off_Attempt_Complete  \
0  20120616144301      Completed         
1  20120616144400      Completed         
2  20120616144700      Completed         
3  20120616144700      Completed         
4  20120616145104      Completed         

                            Off_nibrs_crime_category  \
0  Missing Person / Missing Person               ...   
1  Larceny / Larceny: Misdemeanor                ...   
2  Damage to property / Injury to Real or Persona...   
3  Larceny / Larceny: Misdemeanor                ...   
4  Damage to property / Injury to Real or Persona...   

                            off_nibrs_classification off_noc_code  
0  Missing Person                                ...          800  
1  Theft of Motor Vehicle Parts from Vehicle     ...          23G  
2  All Other Thefts                              ...          290  
3  All Other Thefts                              ...          23H  
4  Damage/Vandalism Of Property                  ...          290

In [26]:
print(Offenses_2012['Off_nibrs_crime_category'].nunique())
print(Offenses_2012['off_nibrs_classification'].nunique())
print(Incident_2012['NIBRS_Hi_Class'].nunique())

119
64
64


Offenses has an additional level of crime classification category with Off_nibrs_crime_category, while Incidents only has nibrs_hi_class. Will keep both because we plan to self-define property crimes.

In [4]:
Property_2012 = pd.read_csv('../theft_and_crime_clt/Data_input/2012/2012_Property.csv')
Property_2012.head()

Complaint_No  Property_No  \
0  20120616144400            1   
1  20120616144700            1   
2  20120616144700            2   
3  20120616145104            1   
4  20120616145203            1   

                                       Property_type  \
0  Vehicle Tag                                   ...   
1  Other                                         ...   
2  Other                                         ...   
3  Dwelling: Single-Family                       ...   
4  Purses/Handbags/Wallets                       ...   

                                       Property_Desc  
0  Vehicles and Parts                            ...  
1  Miscellaneous                                 ...  
2  Miscellaneous                                 ...  
3  Structures                                    ...  
4  Money and Securities                          ...

In [27]:
Property_2012['Property_No'].describe()

count    203577.000000
mean          3.440271
std           6.246846
min           1.000000
25%           1.000000
50%           2.000000
75%           4.000000
max         192.000000
Name: Property_No, dtype: float64

Some crimes have a lot of impacted property items, with the average being 3.44, and the maximum being 192. Since we are focused There may be multiple property instances per crime if more than one thing was damaged or taken -- how do we want to handle this?

In [28]:
Property_2012['Property_Desc'].unique()

array(['Vehicles and Parts                                ',
       'Miscellaneous                                     ',
       'Structures                                        ',
       'Money and Securities                              ',
       'Other Property                                    ',
       'Withheld                                          ',
       'Electronic Equipment and Accessories              ',
       'Jewelry and Collectibles                          ',
       'Consumable Goods                                  ',
       'Building Materials                                ',
       'Electronic Equipment and Accesssories             ',
       'Crops and Landscaping                             ',
       'Animals                                           '], dtype=object)

In [30]:
Property_2012['Property_type'].unique()

array(['Vehicle Tag                                       ',
       'Other                                             ',
       'Dwelling: Single-Family                           ',
       'Purses/Handbags/Wallets                           ',
       'Credit/Debit Cards                                ',
       'Money/Cash                                        ',
       'Windows                                           ',
       'Withheld                                          ',
       'Telephone Equipment                               ',
       'Tools (Power and Hand)                            ',
       'TV Equipment                                      ',
       'Video Game Systems and Accessories                ',
       'Computer Hardware/Software                        ',
       'Camera/Optical Equipment                          ',
       'Doors/Garage Doors                                ',
       'Jewelry                                           ',
       'Vehicle Parts/Ac

Property_desc is a higher level category of property involved, while Property_type gets much more specific

In [5]:
Vehicle_2012 = pd.read_csv('../theft_and_crime_clt/Data_input/2012/2012_Stolen_Vehicle.csv')
Vehicle_2012.head()

Complaint_No Vehicle_Year  \
0  20120616212404         2004   
1  20120617102701         1997   
2  20120617153800         1999   
3  20120617213000         1999   
4  20120617215500         2006   

                                       Vechicle_Make  \
0  Buick                                         ...   
1  Chrysler                                      ...   
2  Chevrolet                                     ...   
3  Oldsmobile                                    ...   
4  Hyundai                                       ...   

                    Vehicle_Model  \
0  CENTURY                          
1  TOWN AND COUNTRY                 
2  CAMARO                           
3  CUTLASS                          
4  SONATA                           

                                       Vehicle_Color  \
0  Silver/Silver                                 ...   
1  White/White                                   ...   
2  Green/Green                                   ...   
3  White/White                                   ...   
4  Tan/Tan                                       ...   

                                        Vehicle_Body  
0  4-Door                                        ...  
1  Van                                           ...  
2  2-Door                                        ...  
3  4-Door                                        ...  
4  4-Door                                        ...

Vehicles are only for stolen vehicles -- do we want to include these and consider them a type of property? Currently not included in property file (above) because they are considered a unique class.

In [6]:
Business_2012 = pd.read_csv('../theft_and_crime_clt/Data_input/2012/2012_Victim_Business.csv')
Business_2012.head()

Complaint_No  Victim_Bus_No  \
0  20120616144700              1   
1  20120616151205              1   
2  20120616155402              1   
3  20120616160600              1   
4                              2   

                                Victim_Bus_LegalName  \
0  COMPRESSOR SERVICE CO.                        ...   
1  THE UNIVERSAL CHURCH                          ...   
2  STATE OF NORTH CAROLINA                       ...   
3  CITGO TIMES INC                               ...   
4  STATE OF NORTH CAROLINA                       ...   

                                     Victim_Bus_Name  \
0  COMPRESSOR SERVICE CO.                        ...   
1  THE UNIVERSAL CHURCH                          ...   
2  STATE OF NORTH CAROLINA                       ...   
3  CITGO TIMES INC                               ...   
4  STATE OF NORTH CAROLINA                       ...   

                                     Victim_Bus_Type Victim_Bus_Block_No  \
0  Business                                      ...                 601   
1  Religious Organization                        ...                5000   
2  Society/Public                                ...                       
3  Business                                      ...                8400   
4  Society/Public                                ...                       

  Victim_Bus_Direction                    Victim_Bus_Street_Name  \
0                   W   24TH                                       
1                   N   TRYON                                      
2                                                                  
3                       OLD STATESVILLE                            
4                                                                  

  Victim_Bus_Suffix Victim_Bus_Apt_No1 Victim_Bus_Apt_No2  \
0                ST                                         
1                ST                                         
2                                                           
3                RD                                         
4                                                           

                            Victim_Bus_City Victim_Bus_State  \
0  CHARLOTTE                                              NC   
1  CHARLOTTE                                              NC   
2                                                              
3  CHARLOTTE                                              NC   
4                                                              

  Victim_Bus_ZipCode  
0              28206  
1                     
2                     
3              28269  
4

In [17]:
Business_2012['Victim_Bus_Type'].nunique()

129

In [16]:
Incident_2012['Place2'].nunique()

68

In [18]:
Incident_2012['Place1'].nunique()

5

Business includes the name of the business impacted (or State of NC for public crimes) as well as a more specific classification of business type with 129 unique values for Victim_Bus_Type rather than the 68 in Place2 of Incident.

In [19]:
Person_2012 = pd.read_csv('../theft_and_crime_clt/Data_input/2012/2012_Victim_Person.csv')
Person_2012.head()

Complaint_No  Victim_No Victim_Age                     Victim_Race  \
0  20120616144301          1         50  Black                            
1  20120616144400          1         26  White                            
2  20120616145104          1         35  Black                            
3  20120616145201          1         26  Black                            
4  20120616145203          1         62  White                            

  Victim_Gender                                   Victim_Ethnicity  
0             M  Other/Not Listed                              ...  
1             F  Unknown/Refused                               ...  
2             F  Other/Not Listed                              ...  
3             M  Unknown/Refused                               ...  
4             F  Other/Not Listed                              ...

Victim is personal information about the victim. Will not be needed.

In [20]:
Weapon_2012 = pd.read_csv('../theft_and_crime_clt/Data_input/2012/2012_Weapons.csv')
Weapon_2012.head()

Complaint_No                               Weapon_Type
0  20120616155200  Personal Weapons (hands,feet,teeth,etc.)
1  20120616163800  Personal Weapons (hands,feet,teeth,etc.)
2  20120616163800  Other Weapon                            
3  20120616165800  Personal Weapons (hands,feet,teeth,etc.)
4  20120616170401  Personal Weapons (hands,feet,teeth,etc.)

Weapon file includes information about weapons will not be needed.

Final merging plan: Keep all information from incident to start, then add in Off_nibrs_crime_category from Offenses. Because this is about property, we will merge left with all property information on crime_no, leaving us with a row for each item impacted by a crime.

If we want to include vehicles, we can also concat all of the vehicle thefts and their relavant incidents, creating a new Property_Desc and Property_type of Vehicle (noting here that automobile is already a category in Property_type should be investigated)

For Victime_Bus, pull in business names as well as the more specific Victime_Bus_Type

Do not pull in anything from victim or weapon.